assets/app/view/game/part/assignments.rb

```diff
@@ -34,29 +34,84 @@ def preferred_render_locations
 
         def load_from_tile
           @assignments = @tile.hex&.assignments || {}
+          @assignments_to_show = []
+          stack_group_count = {}
+          stack_group_img = {}
+          @assignments.keys.each do |assignment|
+            stack_group = @game.assignment_stack_group(assignment)
+            img = @game.assignment_tokens(assignment, setting_for(:simple_logos, @game))
+
+            stack_group_count[stack_group] = (stack_group_count[stack_group] || 0) + 1 unless stack_group.nil?
+            stack_group_img[stack_group] = img unless stack_group.nil?
+            @assignments_to_show.append({ 'img' => img, 'count' => 1 }) if stack_group.nil?
+          end
+
+          stack_group_count.keys.each do |group|
+            @assignments_to_show.append({ 'img' => stack_group_img[group], 'count' => stack_group_count[group] })
+          end
```

Comment:
You can use the `Hash.each` method to iterate through key/value pairs, instead of extracting an array of keys and then looking up the values in the hash.
```suggestion
          stack_group_count.each do |group, count|
            @assignments_to_show.append({ 'img' => stack_group_img[group], 'count' => count })
          end
```